packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -8,37 +8,52 @@ import 'package:crypto/crypto.dart' show md5;
 
 import 'file_system.dart';
 
-/// A collection of checksums for a set of input files.
+/// Fingerprint of build inputs.
 ///
-/// This class can be used during build actions to compute a checksum of the
-/// build action inputs, and if unchanged from the previous build, skip the
+/// Consists of checksums for a set of input files and the values of a set of
+/// property keys.
+///
+/// This class can be used during build actions to compute the fingerprint of
+/// the build action inputs, and if unchanged from the previous build, skip the
 /// build step. This assumes that build outputs are strictly a product of the
-/// input files.
-class Checksum {
-  Checksum.fromFiles(Set<String> inputPaths) : _checksums = <String, String>{} {
-    final Iterable<File> files = inputPaths.map(fs.file);
+/// input files and properties provided.
+class Fingerprint {
+  Fingerprint.fromInputs({Set<String> filePaths, Map<String, String> properties}) : _fingerprint = <String, String>{} {
```

Comment:
Nit: `filePaths` could just be an `Iterable<String>`, right? Then you wouldn't need to convert the lists to sets in the tests.